basic_network.go

```diff
@@ -934,3 +945,14 @@ func relayerMapKey(strmID string, opcode Opcode) relayerID {
 func msgChansKey(opcode Opcode, key string) string {
 	return fmt.Sprintf("%v|%v", opcode, key)
 }
+
+func (n *BasicVideoNetwork) sendMessageWithRetry(pid peer.ID, strm *BasicOutStream, op Opcode, msg interface{}) error {
+	if err := strm.SendMessage(op, msg); err != nil {
+		newStrm := n.NetworkNode.RefreshOutStream(pid)
+		if err := newStrm.SendMessage(op, msg); err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Where is the `sendMessageWithRetry`  function called? Also, is just two attempts enough? Should we consider adding a few more retries, maybe with some sleep / backoff logic?

Reply:
Total brain fart for me.  Just added all the files that actually uses the retry method.  

I only added a single retry because this is solving a bug where the node doesn't know a connection has been closed by its peer.  I think if `SendMessage` fails multiple times, there is probably a bigger issue going on...